clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -3805,6 +3812,12 @@ void CodeGenFunction::EmitTrapCheck(llvm::Value *Checked,
                                     SanitizerHandler CheckHandlerID) {
   llvm::BasicBlock *Cont = createBasicBlock("cont");
 
+  if (ClSanitizeExpHot) {
+    Checked =
+        Builder.CreateOr(Checked, Builder.CreateCall(CGM.getIntrinsic(
+                                      llvm::Intrinsic::experimental_hot)));
```

Comment:
It might be helpful if there were a remark that fired when one of these were optimized out.

Maybe we could get that with another intrinsic whose semantics are "print the message in a remark if this gets optimized out", and then stick one of those on the appropriate side of the condition, with an explanation on where it came from, and what the threshold was.

Reply:
This would be the moral equivalent of diagnose_if

Reply:
Thanks, Added TODO, will do in followup patches.
In this patch I'd like just switch approach.